tpi/terraform.py

```diff
@@ -143,3 +143,14 @@ def _shell_default(
         port = f":{port}" if port is not None else ""
         cmd.append(f"{user}{host}{port}")
         run(cmd)
+
+    def state_mv(self, source, destination, **kwargs):
+        """Retain an existing remote object but track it as a
+        different resource instance address.
+        """
+        assert source and destination
+
+        name = source.split(".")[-1]
+
+        with self.make_tf(name) as tf:
+            tf.cmd("state mv", f"{source}", f"{destination}")
```

Comment:
Thinking about this some more, it's non-intuitive that most of the TPI functions depend on per-`iterative-machine` workspace directories created in `make_tf` (meaning that `source` and `destination` here should always be `iterative-machine.<...>` names), it makes more sense for TPI to be generic, but that doesn't have to be handled in this PR

see: https://github.com/iterative/tpi/issues/15